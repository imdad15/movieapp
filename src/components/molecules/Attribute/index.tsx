import React, { FunctionComponent } from "react";
import styled from "styled-components";
import theme from "../../../theme";

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  flex-direction: row;
  padding: auto 32px;
`;

const BoldText = styled.label`
  font-weight: bold;
  color: ${theme.palette.darkText};
  margin: 4px;
  min-width: 80px;
`;

const LighterText = styled.label`
  fontweight: 500;
  color: ${theme.palette.darkText};
  margin: 4px;
  text-align: start;
  display: block;
`;

const Attribute: FunctionComponent<Props> = ({ title, value }) => {
  return (
    <Wrapper>
      <BoldText>{title}</BoldText>
      <LighterText>{value}</LighterText>
    </Wrapper>
  );
};

type Props = {
  title: string;
  value: string;
};

export default Attribute;
