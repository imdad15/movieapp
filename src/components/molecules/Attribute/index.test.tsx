import React from "react";
import { render, screen } from "@testing-library/react";
import Attribute from ".";

describe("Search icon tests", () => {
  it("should match snapshot", () => {
    const { container: component } = render(
      <Attribute title="some title" value="some value" />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render the Attribute component with props", () => {
    render(<Attribute title="some title" value="some value" />);
    expect(screen.getByText("some title")).toBeDefined();
    expect(screen.getByText("some value")).toBeDefined();
  });
});
