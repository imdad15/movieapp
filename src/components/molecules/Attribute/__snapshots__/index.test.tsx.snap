// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search icon tests should match snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ cvhpud"
  >
    <label
      class="sc-gsnTZi edyuV"
    >
      some title
    </label>
    <label
      class="sc-dkzDqf gsjAmn"
    >
      some value
    </label>
  </div>
</div>
`;
